crypto/fipsmodule/ml_kem/mlkem/mlkem_native.h

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) 2024-2025 The mlkem-native project authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef MLK_H
+#define MLK_H
+/*
+ * Public API for mlkem-native
+ *
+ * This header defines the public API of a single build of mlkem-native.
+ *
+ * To use this header, make sure one of the following holds:
+ *
+ * - The config.h used for the build is available in the include paths.
+ * - The values of MLK_BUILD_INFO_LVL and MLK_BUILD_INFO_NAMESPACE are set,
+ * reflecting the security level (512/768/1024) and namespace of the build.
+ *
+ * This header specifies a build of mlkem-native for a fixed security level.
+ * If you need multiple builds, e.g. to build a library offering multiple
+ * security levels, you need multiple instances of this header. In this case,
+ * make sure to rename or #undefine the header guard
+ */
+
+#include <stdint.h>
+
+/*************************** Build information ********************************/
+
+/*
+ * Provide security level (MLK_BUILD_INFO_LVL) and namespacing
+ * (MLK_BUILD_INFO_NAMESPACE)
+ *
+ * By default, this is extracted from the configuration used for the build,
+ * but you can also set it manually to avoid a dependency on the build config.
+ */
+
+/* Skip this if MLK_BUILD_INFO_LVL has already been set */
+#if !defined(MLK_BUILD_INFO_LVL)
+
+/* Option 1: Extract from config */
+#if defined(MLK_CONFIG_FILE)
+#include MLK_CONFIG_FILE
+#else
+#include "config.h"
+#endif
+
+#if MLKEM_K == 2
+#define MLK_BUILD_INFO_LVL 512
+#elif MLKEM_K == 3
+#define MLK_BUILD_INFO_LVL 768
+#elif MLKEM_K == 4
+#define MLK_BUILD_INFO_LVL 1024
+#else
+#error MLKEM_K not set by config file
+#endif
+
+#ifndef MLK_NAMESPACE_PREFIX
+#error MLK_NAMESPACE_PREFIX not set by config file
+#endif
+
+#if defined(MLK_MULTILEVEL_BUILD_WITH_SHARED) || \
+    defined(MLK_MULTILEVEL_BUILD_NO_SHARED)
+#define MLK_BUILD_INFO_CONCAT3_(x, y, z) x##y##_##z
+#define MLK_BUILD_INFO_CONCAT3(x, y, z) MLK_BUILD_INFO_CONCAT_(x, y, z)
```

Comment:
```suggestion
#define MLK_BUILD_INFO_CONCAT3(x, y, z) MLK_BUILD_INFO_CONCAT3_(x, y, z)
```
Is this used? How (if the suggestion is correct and it was missing `3`?)

Reply:
Thank you, Nevine! I opened https://github.com/pq-code-package/mlkem-native/issues/916.

The TLDR here is: `mlkem_native.h` is a header that is only expected to be used if you consume mlkem-native as a library and don't want to pull in `kem.h` and its dependency chain. It is not necessary for AWS-LC since all mlkem-native sources are embedded in `ml_kem.c` and the top-level API available in that source file.

While I think there is benefit to not modifying the import, in this instance, I believe `import.sh` should remove `mlkem/mlkem_native.h` because the use case it serves is not relevant to AWS-LC.  I will adjust the PR accordingly, and follow-up on https://github.com/pq-code-package/mlkem-native/issues/916 separately with @mkannwischer.

Reply:
Fixed in the latest update.

Reply:
@nebeid The latest commit has reintroduced `mlkem_native.h`. Here's why: When https://github.com/aws/aws-lc/pull/2263 is merged and we want to implement the same safety checks here, we need access to the key sizes that were previously obtained from the now-removed `mlkem_params` structs. `mlkem_native.h` provides those key sizes.